### ISSUE TITLE
Geordi can't read Capistrano 3 deployment configuration

### ISSUE THREAD

**Issue by codener** [makandra/geordi#8]
Capistrano 3 changes its API, thereby breaking Geordi's Capistrano Config interpretation.

**Comment by bitboxer** [comment]
I looked into Capistrano 3 yesterday. It now uses rake for all things. Which is a great design decision. Sadly this makes it really really hard to get the configuration data out of it.

Possible solutions for this now are:

* Create a small DSL that can parse the Capistrano config files
* Move the config out of the Capistrano config into something separate and write a geordi-capistrano gem that makes that data accessible inside of Capistrano

What do you think?

**Comment by henning-koch** [comment]
I believe it would be quite straightforward to create a DSL parser that grocks a deployment stage's config file:

```
set :stage, :production

set :deploy_to, '/opt/app'
set :rails_env, 'production'
set :branch, 'production'

server 'web1.host.tld', user: 'deploy', roles: %w(app web db)
server 'web1.host.tld', user: 'deploy', roles: %w(app web cron)
```

**Comment by bitboxer** [comment]
Awesome!! Thanks :+1:

**Comment by henning-koch** [comment]
:heart:

**Comment by bitboxer** [comment]
Can you release this as a new version?

**Comment by codener** [comment]
Sure. However, since Geordi still has very little tests, I'm currently doing a "closed beta" where I use Geordi 1.3 for myself and fix issues as I notice them (five so far).

I plan to release Geordi 1.3 next week.

**Comment by codener** [comment]
@bitboxer I just released Geordi 1.3.

**Comment by bitboxer** [comment]
Awesome. Thanks!

![tumblr_nuu4sshpmw1r09l2vo1_400](https://cloud.githubusercontent.com/assets/56195/12748158/cc020f82-c9a8-11e5-9627-373dabd3a272.gif)
